tests/main/views/test_direct_award.py

```diff
@@ -593,6 +593,114 @@ def test_which_service_did_you_award_page_should_not_render_if_not_locked_render
+        res = client.get(self.url)
+        doc = html.fromstring(res.get_data(as_text=True))
+        return doc.xpath
+
+    @pytest.fixture
+    def data(self):
+        return {
+            'start_date-day': '31',
+            'start_date-month': '12',
+            'start_date-year': '2019',
+            'end_date-day': '1',
+            'end_date-month': '6',
+            'end_date-year': '2021',
+            'buying_organisation': 'Lewisham Council',
+            'value_in_pounds': '100.00'
+        }
+
+    def setup_method(self, method):
+        super().setup_method(method)
+        self.data_api_client.get_direct_award_project.return_value = self._get_direct_award_lock_project_fixture()
+
+    def test_tell_us_about_contract_exists(self, client):
+        assert client.get(self.url).status_code == 200
+
+    def test_tell_us_about_contract_renders(self, xpath):
+        assert xpath('boolean(//h1[contains(normalize-space(), "Tell us about your contract")])')
+
+    def test_tell_us_about_contract_form_fields(self, xpath):
+        assert xpath('count(//input[@type="text"][substring-after(@name, "start_date")])') == 3
+        assert xpath('count(//input[@type="text"][substring-after(@name, "end_date")])') == 3
+        assert xpath('boolean(//input[@type="text"][contains(@name, "value_in_pounds")])')
+        assert xpath('boolean(//input[@type="text"][contains(@name, "buying_organisation")])')
+        assert xpath('boolean(//input[@type="submit"][@value="Submit"])')
+
+    def test_previous_page_button_exists_and_points_to_previous_page(self, xpath):
+        assert xpath(
+            'boolean('
+            '//a[@href="/buyers/direct-award/g-cloud/projects/1/which-service-won-contract"]'
+            '[contains(normalize-space(), "Previous page")])')
+
+    def test_tell_us_about_contract_form_action_url_is_tell_us_about_contract_url(self, xpath):
+        assert xpath('//form[@class="tell-us-about-contract-form"]/@action')[0] == self.url
+
+    def test_tell_us_about_contract_raises_404_if_project_does_not_exist(self, client):
+        self.data_api_client.get_direct_award_project.side_effect = HTTPError(mock.Mock(status_code=404))
+
+        assert client.get('/buyers/direct-award/g-cloud/projects/31415/tell-us-about-contract').status_code == 404
+
+    def test_tell_us_about_contract_raises_404_if_project_is_not_accessible(self, client):
+        self.data_api_client.get_direct_award_project.return_value['project']['id'] = 314159
+        self.data_api_client.get_direct_award_project.return_value['project']['users'][0]['id'] = 321
+
+        assert client.get('/buyers/direct-award/g-cloud/projects/31415/tell-us-about-contract').status_code == 404
+
+    def test_tell_us_about_contract_successful_post_redirects_to_project_overview(self, client, data):
+        res = client.post(self.url, data=data)
+        assert res.status_code == 302
+        assert res.location.endswith('/buyers/direct-award/g-cloud/projects/1')
```

Comment:
I'd include an assert that `self.data_api_client.update_contract_outcome` is called. If that's the right method. Er... we are saving these details to the API, right?

Reply:
Not yet